admin_guide/allocating_node_resources.adoc

```diff
@@ -155,6 +164,107 @@ $ curl <certificate details> https://<master>/api/v1/nodes/cluster.node22/proxy/
 
 See xref:../rest_api/index.adoc[REST API Overview] for more details about certificate details.
 
+[[node-enforcement]]
+== Node enforcement
+
+The node is able to enforce the total amout of compute resources that pods
+may consume based on the configured allocatable value.  This feature significantly
+improves the reliability of the node by preventing end-user pods from starving
+system services (i.e. container runtime, node agent, etc.) of required compute
+resources.  It is strongly encouraged that administrators set aside system
+resources based on the desired density of their workload in order to improve
+system reliability.
+
+The node enforces allocatable constraints via a new cgroup hierarchy
+that enforces quality of service.  All end-user pods are launched in a
+dedicated cgroup hierarchy separate from system daemons.
+
+To configure this ability, the following kubelet arguments are provided.
+
+.Node Cgroup Settings
+====
+[source,yaml]
+----
+kubeletArguments:
+  cgroups-per-qos:
+    - "true" <1>
+  cgroup-driver:
+    - "systemd" <2>
+  enforce-node-allocatable:
+    - "pods" <3>
+----
+<1> Enable or disable the new cgroup hierarchy managed by the node.  Any change
+of this setting requires a full drain of the node.  It is strongly encouraged
+users not change this value.  This flag must be true to allow the node to
+enforce node allocatable.
+<2> The cgroup driver used by the node when managing cgroup hierarchies.  This
+value must match the driver associated with the container runtime.  Valid values
+are "systemd" and "cgroupfs".  The default is "systemd".
+<3> A comma-delimited list of scopes for where the node should enforce node
+allocatable constraints.  Valid values are "pods", "system-reserved", and "kube-reserved".
+The default is "pods" and is not recommended to change at this time.
+====
+
+Optionally, the node can be made to enforce kube-reserved and system-reserved by
+specifying those tokens in the enforce-node-allocatable flag.  If specified, the
+corresponding --kube-reserved-cgroup or --system-reserved-cgroup needs to be provided.
+In future releases, the node and container runtime will be packaged in a common cgroup
+separate from `system.slice`.  Until that time, it is not encouraged for users to
+change the default value of enforce-node-allocatable flag.
+
+System daemons are expected to be treated similar to Guaranteed pods. System daemons
+can burst within their bounding control groups and this behavior needs to be managed
+as part of cluster deployments.  Enforcing system-reserved reservations
+can lead to critical system services being CPU starved or OOM killed on the node. The
+recommendation is to enforce system-reserved only if operators have profiled their nodes
+exhaustively to come up with precise estimates and are confident in their ability to 
+recover if any process in that group is oom_killed.
+
```

Comment:
Might want to add a warning about system daemons whose resource demands scale with pod load i.e. the kubelet/node and docker processes